ports/stm32f4/common-hal/analogio/AnalogOut.c

```diff
@@ -34,24 +35,83 @@
 #include "shared-bindings/microcontroller/Pin.h"
 #include "supervisor/shared/translate.h"
 
+#include "common-hal/microcontroller/Pin.h"
+
+#include "stm32f4xx_hal.h"
+
+//DAC is shared between both channels. 
+//TODO: store as struct with channel info, automatically turn it off if unused 
+//on both channels for power save?
```

Comment:
static state like this is totally fine since it's contained to this file. You'll find similar in the atmel-samd port for timer tracking (pwm).

Reply:
Ya, turning it off when both are deinit would be nice but not critical.

Reply:
I'll add it as an issue and handle during cleanup.